src/transformStateWithClones.js

```diff
@@ -7,7 +7,54 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = Object.assign({}, state);
+  const history = [];
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        addProperties(stateCopy, obj.extraData);
+        break;
+      case 'removeProperties':
+        removeProperties(stateCopy, obj.keysToRemove);
+        break;
+      case 'clear':
+        clearObject(stateCopy);
+        break;
+      default:
+        history.push('1');
```

Comment:
The default case in the switch statement pushes the string '1' to the history. This might not be the intended behavior, as it doesn't represent a state change. Consider pushing a copy of the state or handling this case differently if needed.